tests/virt_autotest/libvirt_virtual_network_init.pm

```diff
@@ -67,8 +67,8 @@ sub run_test {
         upload_logs "/tmp/$guest.xml";
         #Used with attach-detach(hotplugging) interface to confirm all virtual network mode
         #NOTE: Required all guests keep running status
-        #Check that all guests are still running before virtual network tests
-        script_retry("nmap $guest -PN -p ssh | grep open", delay => 30, retry => 6, timeout => 180);
+        #Ensures the given guests is started and fixes some common network issues
+        ensure_online($guest);
```

Comment:
Thanks for the PR, Leon. This replacement should be able to continue test when guest is down/stuck unexpectedly.

 As part of the original intention for the ticket, we need to find root cause why the guest ssh port check fails. I did not find your analysis result in the progress ticket. Do I miss it somewhere? BTW, it will be helpful to add more debug info to let us know the real status of the guest at that moment, before trying to restart the guest in ensure_online. HDYT?

Reply:
> Thanks for the PR, Leon. This replacement should be able to continue test when guest is down/stuck unexpectedly.
> 
> As part of the original intention for the ticket, we need to find root cause why the guest ssh port check fails. I did not find your analysis result in the progress ticket. Do I miss it somewhere? 
Good point,  emm.. sorry for my mistake, I have missed that.  back to [poo#122080](https://progress.opensuse.org/issues/122080), after did more investigation, figure out that there just only reproduce this libvirt_virtual_network_init test failure one time for 15-SP5 FV guest system on 15-SP5 KVM host with build61.1. meanwhile, refer to the current collect logs, confirm that TIMEOUT(i.e 180 sec) to ensure ssh port for the given 15-SP5 FV guest. meanwhile, and rerun this test get PASS results for libvirt_virtual_network_init - https://openqa.suse.de/tests/10189373#step/libvirt_virtual_network_init/1

> BTW, it will be helpful to add more debug info to let us know the real status of the guest at that moment, before trying to restart the guest in ensure_online. HDYT?
yes, this is great idea. currently,  use with ensure_online() for given guest system, which will included that check if we can ping with the given guest system, check with the ssh service connection status for the given guest system,  ensure the default route is set for the given guest system, check if we can ping SLE VM host from the guest and check also if the name resolution works for the given guest system as the default debug info. So, to be honest, I think that is enough debug info to let us know the real status of the guest at that moment during call with ensure_online() with the given guest system.

Reply:
> > BTW, it will be helpful to add more debug info to let us know the real status of the guest at that moment, before trying to restart the guest in ensure_online. HDYT?
> 
> yes, this is great idea. currently, use with ensure_online() for given guest system, which will included that check if we can ping with the given guest system, check with the ssh service connection status for the given guest system, ensure the default route is set for the given guest system, check if we can ping SLE VM host from the guest and check also if the name resolution works for the given guest system as the default debug info. So, to be honest, I think that is enough debug info to let us know the real status of the guest at that moment during call with ensure_online() with the given guest system.

Well, I was talking something different. Maybe let me rephrase... Ensure_online() will give the guest state for ping/ssh service/route/etc. That's true. But it is the state of the VM AFTER it's restarted if not up. But before ensure_online, at this module of libvirt_virtual_network_init, the vms SHOULD BE UP AND RUNNING WELL , for which the original script_retry to check ssh port open is enough. Should not we find out why the vm is NOT in that expected state at this point and eliminate that from happening again? But if we switch to ensure_online , the issue will be hided/workarounded without any warning given. I hope the difference is  stated clearly this time?

Reply:
Great! make sense totally as your mentioned. 
Thank you so much for your great help and suggestion. it is so helpful to me. emm, let me think about it and update code later.

Reply:
@alice-suse short update, we need to ensure the given guest system is keeping **running** status, the **SSH** connection and the **ICMP PING** responses is workable together at the libvirt_virtual_network_init test module as our expected. So, add the new function `get_guest_status()` to ensure the each check point as our expected for given guest system at the libvirt_virtual_network_init test module. 
Meanwhile, update the latest validation run as below:
[gi-guest_developing-on-host_developing-kvm](https://openqa.suse.de/tests/10312928#)